### PR TITLE
Add DB options to Keycloak CR

### LINKED ISSUE #14374: Add DB options to Keycloak CR
Mapped options:
* db
* db-password (as a Secret)
* db-username (as a Secret)
* db-pool-initial-size
* db-pool-max-size
* db-pool-min-size
* db-url
* db-url-database
* db-url-host
* db-url-port
* db-url-properties

CR structure:
```yaml
apiVersion: k8s.keycloak.org/v2alpha1
kind: Keycloak
metadata:
  name: example-kc
spec:
  db:
    vendor: postgres
    usernameSecret: # plain text value not supported by the type
      name: keycloak-db-secret
      key: username
    passwordSecret:
      name: keycloak-db-secret
      key: password
    url: jdbc:postgresql://db-postgres-service/keycloak
    urlDatabase: keycloak
    urlHost: db-postgres-service
    urlPort: 5432
    urlProperties: extrarg
    schema: some-schema
    poolInitialSize: 1 # integer
    poolMinSize: 1 # integer
    poolMaxSize: 3 # integer
```

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #14374

### CODE REVIEW

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/DatabaseSpec.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    @JsonPropertyDescription("The initial size of the connection pool.")
+    private Integer poolInitialSize;
+
+    @JsonPropertyDescription("The minimal size of the connection pool.")
+    private Integer poolMinSize;
+
+    @JsonPropertyDescription("The maximum size of the connection pool.")
+    private Integer poolMaxSize;
+
```
Comment (mabartos) [inline operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/DatabaseSpec.java:61]: So, the `db-url-properties` will be omitted for now?

**Review by mabartos — CHANGES_REQUESTED** [review]
@pedroigor Thanks for the PR! Looks good. I've added a few minor comments.

Moreover, is it possible to also add a test case to the `KeycloakDistConfiguratorTest`, where we can verify all env vars are properly set?

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/DatabaseSpec.java (hunk 1 above)
Comment (pedroigor) [reply operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/DatabaseSpec.java:61]: I think we should and if needed, people can use the `url` option.

**Comment by pedroigor** [comment]
@mabartos Added the test to env vars.

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/DatabaseSpec.java (hunk 1 above)
Comment (mabartos) [reply operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/DatabaseSpec.java:61]: ok, agree

**Comment by mabartos** [comment]
@pedroigor The operator tests failed. You have to fix `WatchedSecretsTest` regarding the username/password secrets.

**Review by vmuzikar — COMMENTED** [review]
@pedroigor Thanks for the PR!

The test failures seem to be related to this PR.

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/KeycloakSpec.java [hunk 2]
```diff
@@ -73,6 +75,10 @@ public class KeycloakSpec {
     @JsonPropertyDescription("In this section you can find all properties related to the settings of transaction behavior.")
     private TransactionsSpec transactionsSpec;
 
+    @JsonProperty("db")
+    @JsonPropertyDescription("In this section you can find all properties related to connect to a database.")
```
Comment (vmuzikar) [inline operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/KeycloakSpec.java:79]:
```suggestion
    @JsonPropertyDescription("In this section you can find all properties related to connecting to a database.")
```

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/DatabaseSpec.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+package org.keycloak.operator.crds.v2alpha1.deployment.spec;
+
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+
+import io.fabric8.kubernetes.api.model.SecretKeySelector;
+import io.sundr.builder.annotations.Buildable;
+
+@Buildable(editableEnabled = false, builderPackage = "io.fabric8.kubernetes.api.builder")
+public class DatabaseSpec {
```
Comment (vmuzikar) [inline operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/DatabaseSpec.java:26]: Should we also add mapping for `db-url-properties`?

Comment (mabartos) [reply operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/DatabaseSpec.java:26]: https://github.com/keycloak/keycloak/pull/14989#discussion_r999290844

Comment (vmuzikar) [reply operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/DatabaseSpec.java:26]: Thanks @mabartos! I missed that...

**Review by vmuzikar — COMMENTED** [review]
Changes LGTM. Let's wait for the tests.

**Review by mabartos — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]

**Pull request merged**